test/e2e/framework/util.go

```diff
@@ -5319,3 +5325,18 @@ func GetClusterZones(c clientset.Interface) (sets.String, error) {
 	}
 	return zones, nil
 }
+
+// WaitForNodeHasTaintOrNot waits for a taint to be added/removed from the node until timeout occurs, whichever comes first.
+func WaitForNodeHasTaintOrNot(c clientset.Interface, nodeName string, taint *v1.Taint, wantTrue bool, timeout time.Duration) error {
+	if err := wait.PollImmediate(Poll, timeout, func() (bool, error) {
+		has, err := NodeHasTaint(c, nodeName, taint)
+		if err != nil {
+			Logf("Failed to check taint %s on node %s or not", taint.ToString(), nodeName)
+			return false, nil
```

Comment:
Isn't this supposed to return erroror nil alone? Also logging and returning the same error makes it difficult to debug is what I feel, just either log or return the error and in the calling function log directly.

Reply:
@ravisantoshgudimetla that return `false, nil` is the return values of the 3rd parameter of closure `wait.PollImmediate()`.

And why it's printing an error is b/c that's an interim error - need to wait for `timeout` time and then return the final error.

Reply:
Ohh inlining, github UI confused me, thanks.